#! /bin/sh /usr/share/dpatch/dpatch-run
## opt-42_epgsearch.dpatch by Christian Wieninger <cwieninger@gmx.de>
## From the vdr-epgsearch plugin: http://www.cwieninger.de.vu
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: This patch allows the epgsearch plugin to replace the schedule
## DP: menu with it's own version.

@DPATCH@
diff -urNad vdr-1.4.2~/menu.c vdr-1.4.2/menu.c
--- vdr-1.4.2~/menu.c	2006-08-30 19:06:47.000000000 +0200
+++ vdr-1.4.2/menu.c	2006-08-30 19:06:47.000000000 +0200
@@ -3130,7 +3130,11 @@
   // Initial submenus:
 
   switch (State) {
-    case osSchedule:   AddSubMenu(new cMenuSchedule); break;
+    case osSchedule:   {
+			cPlugin *p = cPluginManager::GetPlugin("epgsearch");
+    			(p && !p->SetupParse("IsOrgSchedule", "0")) ? AddSubMenu((cOsdMenu *)p->MainMenuAction()) : AddSubMenu(new cMenuSchedule);
+		       }			
+		       break;
     case osChannels:   AddSubMenu(new cMenuChannels); break;
     case osTimers:     AddSubMenu(new cMenuTimers); break;
     case osRecordings: {
@@ -3377,7 +3381,16 @@
   HadSubMenu |= HasSubMenu();
 
   switch (state) {
-    case osSchedule:   return AddSubMenu(new cMenuSchedule);
+    case osSchedule:   {
+			cPlugin *p = cPluginManager::GetPlugin("epgsearch");
+            		if (p && !p->SetupParse("IsOrgSchedule", "0")) {
+			    return AddSubMenu((cOsdMenu *)p->MainMenuAction());
+			    state = osEnd;
+			    }
+			else 
+			    return AddSubMenu(new cMenuSchedule);
+		       }			
+		       break;
     case osChannels:   return AddSubMenu(new cMenuChannels);
     case osTimers:     return AddSubMenu(new cMenuTimers);
     case osRecordings: {
